Dockerfile

```diff
@@ -0,0 +1,45 @@
+# Shared layer with dependencies
+FROM alpine:3.20.1 AS shared
+
+## Install shared dependencies
```

Comment:
It is good practice to combine the `RUN` commands into on single command in order to not create unnecessary layers in the final images. As the `shared` image is also being used by the `final` image, it would make sense to combine them:

```dockerfile
RUN apk add --no-cache openssl make lua5.1 pcre2
```

Also, `apk update` is not needed here: when called with the `--no-cache` option, `apk add` does not make use of any locally cached information (as provided by `apk update`) but instead fetches all required data from the repo. 😊